sdk/test/metrics/meter_test.cc

```diff
@@ -70,6 +71,55 @@ TEST(MeterTest, BasicAsyncTests)
     }
     return true;
   });
+  observable_counter->RemoveCallback(asyc_generate_measurements, nullptr);
+}
+
+constexpr static unsigned MAX_THREADS       = 25;
+constexpr static unsigned MAX_ITERATIONS_MT = 1000;
+
+TEST(MeterTest, StressMultiThread)
+{
+  MetricReader *metric_reader_ptr = nullptr;
+  auto meter                      = InitMeter(&metric_reader_ptr, "stress_test_meter");
+  std::atomic<unsigned> threadCount(0);
+  size_t numIterations = MAX_ITERATIONS_MT;
+  std::atomic<bool> do_collect{false}, do_sync_create{true}, do_async_create{false};
+  std::vector<nostd::shared_ptr<opentelemetry::metrics::ObservableInstrument>>
+      observable_instruments;
+  while (numIterations--)
+  {
+    for (size_t i = 0; i < MAX_THREADS; i++)
+    {
+      if (threadCount++ < MAX_THREADS)
+      {
+        auto t = std::thread([&]() {
+          std::this_thread::yield();
+          if (do_sync_create.exchange(false))
+          {
+            std::string instrument_name = "test_couter_" + std::to_string(numIterations);
+            meter->CreateLongCounter(instrument_name, "", "");
+            do_async_create.store(true);
+          }
+          if (do_async_create.exchange(false))
+          {
+            auto observable_instrument =
+                meter->CreateLongObservableGauge("test_gauge" + std::to_string(numIterations));
```

Comment:
nit: to be consistent?
```suggestion
                meter->CreateLongObservableGauge("test_gauge_" + std::to_string(numIterations));
```

Reply:
Thanks fixed.